src/engine/Actor.ts

```diff
@@ -702,8 +702,8 @@ export class ActorImpl extends Class implements IActionable, IEvented, IPointerE
       if (this.scene) {
          this._prekill(this.scene);
          this.emit('kill', new KillEvent(this));
-         this.scene.remove(this);
          this._isKilled = true;
+         this.scene.remove(this);
```

Comment:
What's the impact of this?

Reply:
this needed to be moved in order for there not to be an infinite loop of
scene-> actor.kill()
actor->scene.remove(actor)
scene->actor.kill()

By moving this I can easily check if the actor has been killed before trying to kill it again in the scene.